In [5]:
import json

table_template = """
| 方法 | 挂钟时间 (秒) | Forward 次数 | 生成 Token 数 | Token 输出速率 (token/秒) | 通信时间 (秒) | Edge-Cloud 数据传输量 (MB) | 通信能量 (焦耳) |
|------|-------------|--------------|--------------|------------------------|-------------|--------------------------|---------------|
| **Dist Split Spec** | {DSSD_wall_time} | {DSSD_forward_times} | {DSSD_generated_tokens} | {DSSD_throughput} | {DSSD_communication_time} | {DSSD_edge_cloud_data_mb} | {DSSD_comm_energy} |
| **Dist Spec** | {DSD_wall_time} | {DSD_forward_times} | {DSD_generated_tokens} | {DSD_throughput} | {DSD_communication_time} | {DSD_edge_cloud_data_mb} | {DSD_comm_energy} |
| **Uncertainty Decoding** | {UD_wall_time} | {UD_forward_times} | {UD_generated_tokens} | {UD_throughput} | {UD_communication_time} | {UD_edge_cloud_data_mb} | {UD_comm_energy} |
| **Tridecoding** | {TD_wall_time} | {TD_forward_times} | {TD_generated_tokens} | {TD_throughput} | {TD_communication_time} | {TD_edge_cloud_data_mb} | {TD_comm_energy} |
"""

file_path = "experiment_summary_20250919_112050.json"

with open(file_path, "r") as f:
    data = json.load(f)

dssd_data, dsd_data, ud_data, td_data = {}, {}, {}, {}

for item in data:
    method = item["exp_name"]
    if "dist_split_spec" in method:
        dssd_data = item
    elif "dist_spec" in method:
        dsd_data = item
    elif "uncertainty_decoding" in method:
        ud_data = item
    elif "tridecoding" in method:
        td_data = item

def bytes_to_mb(val):
    return round(val / 1024 / 1024, 2) if isinstance(val, (int, float)) else "N/A"

table = table_template.format(
    DSSD_wall_time=dssd_data["result"].get("wall_time", "N/A"),
    DSSD_forward_times=dssd_data["result"].get("target_forward_times", "N/A"),
    DSSD_generated_tokens=dssd_data["result"].get("generated_tokens", "N/A"),
    DSSD_throughput=dssd_data["result"].get("throughput", "N/A"),
    DSSD_communication_time=dssd_data["result"].get("communication_time", "N/A"),
    DSSD_edge_cloud_data_mb=bytes_to_mb(dssd_data["result"].get("edge_cloud_data_bytes", "N/A")),
    DSSD_comm_energy=dssd_data["result"].get("comm_energy", "N/A"),
    DSD_wall_time=dsd_data["result"].get("wall_time", "N/A"),
    DSD_forward_times=dsd_data["result"].get("target_forward_times", "N/A"),
    DSD_generated_tokens=dsd_data["result"].get("generated_tokens", "N/A"),
    DSD_throughput=dsd_data["result"].get("throughput", "N/A"),
    DSD_communication_time=dsd_data["result"].get("communication_time", "N/A"),
    DSD_edge_cloud_data_mb=bytes_to_mb(dsd_data["result"].get("edge_cloud_data_bytes", "N/A")),
    DSD_comm_energy=dsd_data["result"].get("comm_energy", "N/A"),
    UD_wall_time=ud_data["result"].get("wall_time", "N/A"),
    UD_forward_times=ud_data["result"].get("target_forward_times", "N/A"),
    UD_generated_tokens=ud_data["result"].get("generated_tokens", "N/A"),
    UD_throughput=ud_data["result"].get("throughput", "N/A"),
    UD_communication_time=ud_data["result"].get("communication_time", "N/A"),
    UD_edge_cloud_data_mb=bytes_to_mb(ud_data["result"].get("edge_cloud_data_bytes", "N/A")),
    UD_comm_energy=ud_data["result"].get("comm_energy", "N/A"),
    TD_wall_time=td_data["result"].get("wall_time", "N/A"),
    TD_forward_times=td_data["result"].get("target_forward_times", "N/A"),
    TD_generated_tokens=td_data["result"].get("generated_tokens", "N/A"),
    TD_throughput=td_data["result"].get("throughput", "N/A"),
    TD_communication_time=td_data["result"].get("communication_time", "N/A"),
    TD_edge_cloud_data_mb=bytes_to_mb(td_data["result"].get("edge_cloud_data_bytes", "N/A")),
    TD_comm_energy=td_data["result"].get("comm_energy", "N/A"),
)

print(table)


| 方法 | 挂钟时间 (秒) | Forward 次数 | 生成 Token 数 | Token 输出速率 (token/秒) | 通信时间 (秒) | Edge-Cloud 数据传输量 (MB) | 通信能量 (焦耳) |
|------|-------------|--------------|--------------|------------------------|-------------|--------------------------|---------------|
| **Dist Split Spec** | 614.8236771930765 | 5327 | 20480 | 33.310363214213936 | 3.759644722373352 | 2.54 | 1.879822361186676 |
| **Dist Spec** | 683.575611573428 | 5327 | 20480 | 29.960109245062043 | 54.13051904413112 | 36.6 | 27.06525952206556 |
| **Uncertainty Decoding** | 677.6719210541555 | 577 | 20480 | 30.221113438110653 | 3.8512530854054905 | 26.04 | 1.9256265427027452 |
| **Tridecoding** | 882.8735454999838 | 4605 | 21305 | 24.131428683747316 | 6.633795622054229 | 3.9 | 3.316897811027124 |





| 方法 | 挂钟时间 (秒) | Forward 次数 | 生成 Token 数 | Token 输出速率 (token/秒) | 通信时间 (秒) | Edge-Cloud 数据传输量 (MB) | 通信能量 (焦耳) |
|------|-------------|--------------|--------------|------------------------|-------------|--------------------------|---------------|
| **Dist Split Spec** | 874.9531330036236 | 5327 | 20480 | 23.406968016325948 | 268.83964472237307 | 2.54 | 134.41982236118653 |
| **Dist Spec** | 854.8715451671778 | 5327 | 20480 | 23.956815635961952 | 234.62051904413104 | 36.6 | 117.31025952206552 |
| **Uncertainty Decoding** | 991.2522008393123 | 577 | 20480 | 20.66073596876677 | 316.9112530854054 | 26.04 | 158.4556265427027 |
| **Tridecoding** | 1339.6341556074049 | 4605 | 21305 | 15.903595702470037 | 451.6587956220537 | 3.9 | 225.82939781102795 |
